src/json/next/serialization.cr

```diff
@@ -2,6 +2,109 @@ module JSON
   annotation Field
   end
 
+  # The `JSON::Serializable` module is a better alternative to `JSON.mapping`.
```

Comment:
I'd rather not refer to this as a replacement for `JSON.mapping`, since `JSON.mapping` will probably go away before 1.0 if this works out. Better to provide a full description of what this code attempts to do, which is provide automatic serialization of the object to/from json.